apps/_pomodoro_app/realtime/application/timer.js

```diff
@@ -0,0 +1,107 @@
+const CyclicQueue = require('./utils/cyclicQueue');
+
+const status = {
+    'paused':'paused',
+    'working':'working',
+}
+
+const periods = {
+    'work': 'WORK',
+    'break': 'BREAK',
+    'bigBreak':'BIG_BREAK'
+}
+
+function getPeriodQueue(breakTime, bigBreakTime, worktimeTime) {
+
+    const workInterval = new Period(periods.work, worktimeTime)
+    const breakInterval = new Period(periods.break, breakTime)
+    const bigBreakInterval = new Period(periods.bigBreak, bigBreakTime)
+
+    return new CyclicQueue([
+        workInterval,
+        breakInterval,
+        workInterval,
+        breakInterval,
+        workInterval,
+        breakInterval,
+        workInterval,
+        bigBreakInterval
+    ])
+}
+
+/**
+ * Period model
+ */
+class Period {
+    constructor(name, time) {
+        this.name = name
+        this.time = time
+    }
+}
+
+/**
+ * Sets timers off, gives away current time, is unique for one team
+ */
+class Timer {
+
+    constructor(breakTime, bigBreakTime, worktimeTime) {
+        this.timer = 0
+        this.counterFunction = 0
```

Comment:
`counterFunctionId`

Reply:
Would not agree on that: this variable holds the timer function, it doesn't encode the actual ID of the function

Reply:
I think the 'undefined' value as default if more clear and predictable rather than 0.

It's interesting. Usually, the return value is named with ID suffix (ex: https://javascript.info/settimeout-setinterval#setinterval) it's common style. But, the returning value for Node.js is more complex %) 
![image](https://user-images.githubusercontent.com/1394025/86594800-18897680-bfb1-11ea-905f-78e547f9ae2a.png)